spec/requests/v2/plates_spec.rb

```diff
@@ -0,0 +1,153 @@
+            { position: 'H12', name: 'Sample in H12' }
+          ]
+        )
+      end
+
+      it 'returns plate with nested sample' do
+        post v2_path, params: { query: '{ plates { wells { material { ... on Request { sample { name } } } } } }' }
+        expect(response).to have_http_status(:success)
+        json = ActiveSupport::JSON.decode(response.body)
+        expect(json['data']['plates'].length).to eq(1)
+        expect(json['data']['plates'].first).to include(
+          'wells' => [
+            {
+              'material' => {
+                'sample' => {
+                  'name' => 'Sample in A1'
+                }
+              }
+            },
+            {
+              'material' => {
+                'sample' => {
+                  'name' => 'Sample in H12'
+                }
+              }
+            }
+          ]
+        )
+      end
+    end
+  end
+
+  context 'create plate' do
+    def valid_query()
+      <<~GQL
+      mutation {
+        createPlateWithOntSamples(
+          input: {
+            arguments: {
+              barcode: "PLATE-1234"
+              wells: [
+                { position: "A1" sample: { name: "Sample for A1" externalId: "ExtIdA1" } }
+                { position: "E7" sample: { name: "Sample for E7" externalId: "ExtIdE7" } }
+              ]
+            }
+          }
+        )
+        {
+          plate { barcode wells { material { ... on Request { sample { name externalId } } } } }
+          errors
+        }
+      }
+      GQL
+    end
+
+    it 'creates a plate with provided parameters' do
+      post v2_path, params: { query: valid_query }
+      expect(response).to have_http_status(:success)
+      json = ActiveSupport::JSON.decode(response.body)
+
+      mutation_json = json['data']['createPlateWithOntSamples']
+      plate_json = mutation_json['plate']
+      expect(plate_json['barcode']).to eq('PLATE-1234')
+
+      samples_json = plate_json['wells'].map { |well| well['material']['sample'] }
+      expect(samples_json).to contain_exactly({
+        'name' => 'Sample for A1',
+        'externalId' => 'ExtIdA1'
+      },
+      {
+        'name' => 'Sample for E7',
+        'externalId' => 'ExtIdE7'
+      })
+
+      expect(mutation_json['errors']).to be_empty
+    end
+
+    def invalid_data_query()
+      # Wells have no positions
+      <<~GQL
+      mutation {
+        createPlateWithOntSamples(
+          input: {
+            arguments: {
+              barcode: "PLATE-1234"
+              wells: [
+                { sample: { name: "Sample for A1" externalId: "ExtIdA1" } }
+                { sample: { name: "Sample for E7" externalId: "ExtIdE7" } }
+              ]
+            }
+          }
+        )
+        {
+          plate { barcode wells { material { ... on Request { sample { name externalId } } } } }
+          errors
+        }
+      }
+      GQL
+    end
+
+    it 'provides an error when data was invalid' do
```

Comment:
Does this test the response when the request factory returns errors?
If so would it not be better mocking out the factory to return errors irrespective of the input, so you don't have factory logic leaking into these tests?
If not, does this not just test the same thing as the below `provides an error when missing required fields` test, and a new test should be added to test getting errors from the factory?

Reply:
Excellent point, and yes it is testing that we handle errors coming back from the factory.  I'll mock out the error response using the original GraphQL query from the first test.